codex/blockexchange/engine/engine.nim

```diff
@@ -116,39 +116,66 @@ proc stop*(b: BlockExcEngine) {.async.} =
 
   trace "NetworkStore stopped"
 
+proc sendWantHave(b: BlockExcEngine, cid: Cid, selectedPeer: BlockExcPeerCtx, peers: seq[BlockExcPeerCtx]): Future[void] {.async.} =
+  trace "Sending wantHave request to peers", cid
+  for p in peers:
+    if p != selectedPeer:
+      if cid notin p.peerHave:
+        trace " wantHave > ", peer = p.id
+        await b.network.request.sendWantList(
+          p.id,
+          @[cid],
+          wantType = WantType.WantHave) # we only want to know if the peer has the block
+
+proc sendWantHaveSingle(b: BlockExcEngine, cid: Cid, peer: PeerId): Future[void] {.async.} =
+  trace "Sending wantHave request to single peer", cid, peer
+  await b.network.request.sendWantList(
+    peer,
+    @[cid],
+    wantType = WantType.WantHave) # we only want to know if the peer has the block
+
+proc sendWantBlock(b: BlockExcEngine, cid: Cid, blockPeer: BlockExcPeerCtx): Future[void] {.async.} =
+  trace "Sending wantBlock request to", peer = blockPeer.id, cid
+  await b.network.request.sendWantList(
+    blockPeer.id,
+    @[cid],
+    wantType = WantType.WantBlock) # we want this remote to send us a block
+
+proc findCheapestPeerForBlock(b: BlockExcEngine, cheapestPeers: seq[BlockExcPeerCtx]): ?BlockExcPeerCtx =
+  if cheapestPeers.len <= 0:
+    trace "No cheapest peers, selecting first in list"
+    let peers = toSeq(b.peers) # Get any peer
+    if peers.len <= 0:
+      return none(BlockExcPeerCtx)
+    return some(peers[0])
+  return some(cheapestPeers[0]) # get cheapest
+
 proc requestBlock*(
   b: BlockExcEngine,
   cid: Cid,
   timeout = DefaultBlockTimeout): Future[bt.Block] {.async.} =
-  ## Request a block from remotes
-  ##
-
-  trace "Requesting block", cid, peers = b.peers.len
+  trace "Begin block request", cid, peers = b.peers.len
 
   if b.pendingBlocks.isInFlight(cid):
     trace "Request handle already pending", cid
     return await b.pendingBlocks.getWantHandle(cid, timeout)
 
-  let
-    blk = b.pendingBlocks.getWantHandle(cid, timeout)
-
-  var
-    peers = b.peers.selectCheapest(cid)
+  let blk = b.pendingBlocks.getWantHandle(cid, timeout)
```

Comment:
Making this changes everywhere just makes the codebase inconsistent. The codebase is already written in a specific style which we better stick to, at least in the code units where this style is prevalent.

Reply:
I agree. I wish we had some kind of linter.